library/agent/wrapRequire.ts

```diff
@@ -0,0 +1,61 @@
+import * as mod from "module";
+import { ModifyingRequireInterceptor } from "./hooks/ModifyingRequireInterceptor";
+import { defineProperty } from "../helpers/wrap";
+import { Agent } from "./Agent";
+const req = mod.prototype.require;
+
+const interceptors: ModifyingRequireInterceptor[] = [];
```

Comment:
if we can rewrite this as a class, would be nice, so that we can reset state if a new instance is made? Similar to the other hooking

As we run this multiple times during tests